# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mansi-cs/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

I chose this lane because the goal is to help content reviewers decide which pages should be reviewed first. There are many pages to review, so prioritizing them can save time and focus effort on the most important opportunities. This lane combines business decision-making with data analysis and machine learning.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question:
Which pages should be reviewed first for content refresh or improvement?

Decision:
Which pages should the content team spend time reviewing?

Who acts on it?
A content reviewer or editor.

Action:
Review, update, expand, or monitor the highest-priority pages.

Cost of a wrong recommendation:
If a page is incorrectly recommended, reviewer time may be wasted. If an important page is missed, a valuable opportunity may be lost.

Why ML can help:
There are many signals and too many pages to review manually. ML can help rank pages based on patterns in the data.

In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [18]:
!git clone https://github.com/mansi-cs/flyrank-ml-internship.git
%cd flyrank-ml-internship

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (83/83), done.
remote: Total 127 (delta 41), reused 95 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (127/127), 1.84 MiB | 15.61 MiB/s, done.
Resolving deltas: 100% (41/41), done.
/content/flyrank-ml-internship/flyrank-ml-internship


In [19]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print(df["trend_direction"].value_counts())

Rows: 30000
Columns: 44
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30,000 content items and 44 features, which provides enough data to investigate content refresh opportunities. Looking at trend_direction, 16,262 pages are trending down, while only 4,388 are trending up.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is intended to support content review decisions. It may identify pages that appear to have refresh or improvement opportunities based on observed search and engagement signals. However, it cannot prove that updating a page will cause better performance. The results should be treated as decision-support recommendations, not guarantees. I will make observational and directional claims only and will not claim to predict algorithms or prove causation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.